### PR TITLE
nrf9160: samples: aws_fota: Reboot after modem DFU

### PR DESCRIPTION
**Pull request by simensrostad**
Reboot after modem DFU.

If a modem DFU has been performed, a reboot is expected to fully
apply the new modem firmware image.

Closes CIA-344

### CODE REVIEW

**Review by rlubos — APPROVED** [review]

File: samples/nrf9160/aws_fota/src/main.c [hunk 1]
```diff
@@ -403,27 +403,28 @@ void main(void)
...
 	switch (err) {
+	case 0:
+		/* Initialization successful, no action required. */
+		break;
 	case MODEM_DFU_RESULT_OK:
 		printk("Modem firmware update successful!\n");
 		printk("Modem will run the new firmware after reboot\n");
-		k_thread_suspend(k_current_get());
-		break;
+		sys_reboot(SYS_REBOOT_COLD);
```
Comment (nvlsianpu) [inline samples/nrf9160/aws_fota/src/main.c:412]: I'm not sure whether the printk message above will be transmitted completely. Can you check and add k_sleep() if needed.

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: samples/nrf9160/aws_fota/src/main.c (hunk 1 above)
Comment (simensrostad) [reply samples/nrf9160/aws_fota/src/main.c]: The last line wasn't printed. Added a small delay, thanks!

**Comment by simensrostad** [comment]
@nvlsianpu Updated the PR per your suggestion. Please revisit.

**Review by nvlsianpu — APPROVED** [review]

**Comment by jayteemo** [comment]
Perhaps too late and not a big deal, but I believe your JIRA ticket reference is incorrect.

**Comment by simensrostad** [comment]
> Perhaps too late and not a big deal, but I believe your JIRA ticket reference is incorrect.

Yes, you're right. I should be CIA-433. 😧

**Pull request merged**